ogre2/include/ignition/rendering/ogre2/Ogre2RenderTarget.hh

```diff
@@ -134,10 +137,6 @@ namespace ignition
       /// \brief Update the render pass chain
       protected: virtual void UpdateRenderPassChain();
 
-      /// \brief Create a compositor shadow node with the same number of shadow
-      /// textures as the number of shadow casting lights
-      protected: void UpdateShadowNode();
```

Comment:
We'll need to deprecate this function first before removing it since it's exposed to the users. Can you mark this function as deprecated?

```
protected: void IGN_DEPRECATED(5) UpdateShadowNode();
```

The implementation can just be a warning msg, something like:

```
ignwarn << "Ogre2RenderTarget::UpdateShadowNode() is deprecated and replaced by Ogre2Scene:UpdateShadowNode()" << std::endl;
```

Reply:
> One issue I noticed is that when I try to add a new shadow-casting light (to trigger a rebuild of compositor), ign gazebo crashes. This can be reproduced with the following steps:

Cannot reproduce. Adding new lights works perfectly fine for me. I'm even trying on full debug. Do you have a callstack?

Perhaps something didn't re-build cleanly, since as you said the ABI changed and if a component wasn't properly rebuilt, at some point the wrong virtual function(s) will be called.

Reply:
Nevermind. Repro found.

Reply:
Well I am shocked!

We hit an Ogre 2.1 bug that was fixed in 2.2; fortunately it's easy to workaround from Gazebo, and that workaround can be removed in 2.2